### PR TITLE
Improve Citrix CTX1

### PR DESCRIPTION
**Pull request by SkeletalDemise**
* Citrix CTX1 decoder now handles subtraction overflows
* Removed uppercase check, now it can decode lowercase Citrix CTX1 strings
* Added 3 more tests
* Removed 2 redundant tests
* Added docs to every test

### CODE REVIEW

File: src/decoders/citrix_ctx1_decoder.rs [hunk 1]
```diff
@@ -37,7 +37,7 @@ impl Crack for Decoder<CitrixCTX1Decoder> {
         let mut results = CrackResult::new(self, text.to_string());
 
         if decoded_text.is_none() {
-            debug!("Failed to decode citrix_ctx1 because the length is not a multiple of 4");
+            debug!("Failed to decode citrix_ctx1");
```
Comment (bee-san) [inline src/decoders/citrix_ctx1_decoder.rs:40]: Anyway we can propagate an error here or something? Why couldn't it decode?

File: src/decoders/citrix_ctx1_decoder.rs [hunk 2]
```diff
@@ -110,78 +109,84 @@ mod tests {
...
+        );
+        assert_eq!(
+            result.unencrypted_text.unwrap(),
+            "This is lowercase Citrix CTX1"
+        );
+    }
+
+    #[test]
+    fn test_citrix_ctx1_substraction_overflows() {
+        // This tests if Citrix CTX1 can handle substraction overflows
```
Comment (bee-san) [inline src/decoders/citrix_ctx1_decoder.rs:137]:
```suggestion
    fn test_citrix_ctx1_handles_substraction_overflow() {
        // This tests if Citrix CTX1 can handle substraction overflows
```

**Review by bee-san — CHANGES_REQUESTED** [review]

File: src/decoders/citrix_ctx1_decoder.rs [hunk 3]
```diff
@@ -110,78 +109,84 @@ mod tests {
 
     #[test]
     fn test_citrix_ctx1() {
+        // This tests if Citrix CTX1 can decode Citrix CTX1 successfully
```
[the remark this answers is not in the document]
Comment (SkeletalDemise) [reply src/decoders/citrix_ctx1_decoder.rs]: Done.

File: src/decoders/citrix_ctx1_decoder.rs [hunk 4]
```diff
@@ -110,78 +109,84 @@ mod tests {
...
         );
         assert_eq!(result.unencrypted_text.unwrap(), "hello world");
     }
+
     #[test]
-    fn citrix_ctx1_decode_empty_string() {
-        // Citrix_ctx1 returns an empty string, this is a valid citrix_ctx1 string
-        // but returns False on check_string_success
+    fn test_citrix_ctx1_lowercase() {
+        // This tests if Citrix CTX1 can decode lowercase strings
```
[the remark this answers is not in the document]
Comment (SkeletalDemise) [reply src/decoders/citrix_ctx1_decoder.rs]: Done.

File: src/decoders/citrix_ctx1_decoder.rs (hunk 2 above)
Comment (SkeletalDemise) [reply src/decoders/citrix_ctx1_decoder.rs]: Done.

File: src/decoders/citrix_ctx1_decoder.rs [hunk 5]
```diff
@@ -110,78 +109,84 @@ mod tests {
...
-            .crack("", &get_athena_checker())
+            .crack("NUWEN43XR44TLAYHSU4DVI2ISF======", &get_athena_checker())
+            .unencrypted_text;
+        assert!(result.is_none());
+    }
+
+    #[test]
+    fn test_citrix_ctx1_length() {
+        // This tests if Citrix CTX1 can handle strings with length that are not divisible by 4
+        // It should return None
```
[the remark this answers is not in the document]
Comment (SkeletalDemise) [reply src/decoders/citrix_ctx1_decoder.rs]: Done.

**Review by bee-san — APPROVED** [review]

**Pull request merged**